src/redis-helper.ts

```diff
@@ -145,6 +207,15 @@ export async function getTopTracksForArtist(redisClient: redis.RedisClient,
             });
         }
 
+        // set the last updated timestamp
```

Comment:
Instead of having a whole separate update here, I'd just add the `top_track_ids_last_updated` key into the update on line 193 above so both `top_track_ids` and the date are updated at the same time

Reply:
My original thinking was that there was still some logic that occurred after the initial IDs being set, but admittedly if something fails in between the data is still half-baked and you'd need to pick it up from errors in the log anyway. Done.